vm-setup/roles/common/tasks/extra_networks_tasks.yml

```diff
@@ -0,0 +1,54 @@
+---
+
+- name: "Processing extra network {{network_name}}"
+  debug:
+    var: network_name
+
+# Get the CIDR for ipv4 and/or ipv6
+- name: "Set {{network_name}}_cidr_v4 from {{network_name.upper()}}_NETWORK_SUBNET_V4"
+  set_fact:
+    "{{network_name}}_cidr_v4": "{{lookup('env', network_name.upper()+'_NETWORK_SUBNET_V4')}}"
+
+- name: "Set {{network_name}}_cidr_v6 from {{network_name.upper()}}_NETWORK_SUBNET_V6"
+  set_fact:
+    "{{network_name}}_cidr_v6": "{{lookup('env', network_name.upper()+'_NETWORK_SUBNET_V6')}}"
+
+# Validate that we have at least one CIDR, and that the format is correct
+- fail:
+    msg: "Must specify at least one of {{network_name.upper()}}_NETWORK_SUBNET_V4 or {{network_name.upper()}}_NETWORK_SUBNET_V6"
+  when: "{{ lookup('vars', network_name + '_cidr_v4') == '' and lookup('vars', network_name + '_cidr_v6') == '' }}"
+
+- name: Calculate v4 extra_network data
+  block:
+    - set_fact:
+        extra_network_v4: "{{ [{
+        'address_v4': cidr|nthhost(1),
+        'netmask_v4': cidr|ipaddr('netmask'),
+        'dhcp_range_v4': [ cidr|nthhost(20), cidr|nthhost(60)],
+        }]}}"
+  when: "{{ lookup('vars', network_name + '_cidr_v4') != '' }}"
```

Comment:
I think we would not reach here in this condition because we will fail earlier in L19, but extra precautions is good to have

Reply:
Actually I missed the fact that it is logical AND in L19, so there is  a chance to come here (example when v6 CIDR is in correct format but not v4) with incorrect v4 CIDR.